test/python/dialects/indexing/dialect.py

```diff
@@ -439,3 +440,37 @@ def testSimpleLiteralIndexing():
     print(Tensor(w.owner.operands[1]))
     # CHECK: %{{.*}} = indexing.gather %[[TEN]][%[[CST0]]] gather_dims([0, 1, 2]) unique : (tensor<10x10x10x10xi32>, tensor<3xindex>) -> tensor<10xi32>
     print(w.owner)
+
+
+# CHECK-LABEL: TEST: testCanonicalizeTupleIndexCastListLiteral
+@run
+def testCanonicalizeTupleIndexCastListLiteral():
```

Comment:
I don't understand what this test is doing. Can you explain?

Reply:
Actually, I think that tests should be so simple that they don't need explanation. Otherwise, you'd need to test the test. Even if this gets quite a bit longer, maybe it is better to spell out the cases here?

Reply:
This test tests whether the canonicalization works for all permutations of non-trivial indexer and `:`, so e.g., 

```
ten[idx, :, :, :]
ten[:, idx, :, :]
ten[:, :, idx, :]
...
```

You get the idea. I did have lots of these cases spelled out before (I used this same code to generate the cases) but I think you can see that it's the same case just permuted. And, since this is a unit test for an unexposed method, and the only thing that needs to be verified is the `assert` (i.e., no IR prints and such), I changed it to this. I can change it back but it will really will blow up the length of this file.

Reply:
This seems fine to me, I would just add comments that you generate and test all possible versions on the fly and add 1-2 examples of MLIR that gets generated.